SleepEngine/Engine/GameTemplate/Game.cpp

```diff
@@ -32,66 +31,45 @@ Game::Game(size_t width, size_t height)
 
     if (m_instance)
     {
-	    LOG_AND_ASSERT_ERROR(false, "Attempt to create second game window! It is forbidden!");
-	    return;
+        LOG_AND_ASSERT_ERROR(false, "Attempt to create second game window! It is forbidden!");
+        return;
     }
     m_instance = this;
     m_renderer = std::make_unique <GameRenderer>();
 }
 
-Game::SceneIDType Game::addScene(SceneIniter initer)
+void Game::addScene(Scene::Initer initer, std::string_view id)
 {
-    auto sceneAndIniter = std::make_pair(Scene(), initer);
-    auto [it, wasInserted] = m_scenes.insert({ m_nextSceneID++, std::move(sceneAndIniter) });
-    assertion(wasInserted, "nextSceneID overflow");
+    auto& scene = m_scenes.emplace_back(std::move(initer), id);
 
     if (m_scenes.size() == 1)
     {
-        m_currentScene = it;
-        initCurrentScene();
+        // first scene
+        scene.init();
```

Comment:
`init` possibly can be a very long operation. Maybe we shouldn't do it automatically here?

Reply:
Yeah, we should do it in game run

Reply:
Fixed